packages/app-store/googlevideo/components/InstallAppButton.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react";
+
+import type { InstallAppButtonProps } from "@calcom/app-store/types";
+import { trpc } from "@calcom/trpc/react";
+import type { DialogProps } from "@calcom/ui";
+import { Button } from "@calcom/ui";
+import { Dialog, DialogClose, DialogContent, DialogFooter } from "@calcom/ui";
+
+import useAddAppMutation from "../../_utils/useAddAppMutation";
+
+export default function InstallAppButton(props: InstallAppButtonProps) {
+  const [showWarningDialog, setShowWarningDialog] = useState(false);
+
+  return (
+    <>
+      {props.render({
+        onClick() {
+          setShowWarningDialog(true);
+          // mutation.mutate("");
+        },
+        disabled: showWarningDialog,
+      })}
+      <WarningDialog open={showWarningDialog} onOpenChange={setShowWarningDialog} />
+    </>
+  );
+}
+
+function WarningDialog(props: DialogProps) {
+  const [googleCalendarPresent, setGoogleCalendarPresent] = useState<undefined | boolean>(undefined);
+
+  const mutation = useAddAppMutation(googleCalendarPresent ? "google_video" : "google_calendar", {
+    installGoogleVideo: !googleCalendarPresent,
+  });
+  const { data, isLoading } = trpc.viewer.integrations.useQuery(
```

Comment:
<img width="954" alt="Screenshot 2023-01-06 at 8 09 45 PM" src="https://user-images.githubusercontent.com/1780212/211034045-ed0d772e-92e0-4434-8487-4f9d7d27a711.png">

I think we can directly use `useApp` here which fetches app by appId. This would void fetching unnecessary calendar apps.